pkg/cmd/util/cmd.go

```diff
@@ -31,3 +33,32 @@ func GetDisplayFilename(filename string) string {
 
 	return filename
 }
+
+// ResolveResourceSyntax will return the kind (if found), name and a possible error
+// from the provided arg. If a default kind has been specified, then arg
+// should respect it.
+func ResolveResourceSyntax(mapper meta.RESTMapper, defaultKind, arg string) (string, string, error) {
+	if mapper == nil {
+		return "", "", errors.New("mapper cannot be nil")
+	}
+
+	var name, kind string
+	parts := strings.Split(arg, "/")
+	switch len(parts) {
+	case 1:
+		name = parts[0]
```

Comment:
What you are asking about is `ResourceNames` in the resource builder. This is different.